src/pages/TeamOverview.tsx

```diff
@@ -62,34 +62,51 @@ interface PageState {
 const TeamOverview = () => {
     const location = useLocation();
     const {teamId} = useParams();
-    const [pageData, setPageData] = React.useState<PageState>({});
-    const [isLoading, setIsLoading] = React.useState<boolean>(true);
-
-    React.useEffect(() => {
-        var getTeamUsers = async () => {
-            const {teamLeadId, teamMemberIds = []} = await getTeamOverview(teamId);
-            const teamLead = await getUserData(teamLeadId);
-
-            const teamMembers = [];
-            for(var teamMemberId of teamMemberIds) {
-                const data = await getUserData(teamMemberId);
-                teamMembers.push(data);
+    const [pageData, setPageData] = useState<PageState>({});
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const [filteredMembers, setFilteredMembers] = useState<UserData[]>([]);
+
+    useEffect(() => {
+        const fetchTeamUsers = async () => {
+            try {
+                const {teamLeadId, teamMemberIds = []} = await getTeamOverview(teamId);
+                const teamLead = await getUserData(teamLeadId);
+
+                const teamMembers = await Promise.all(teamMemberIds.map(getUserData));
+
+                setPageData({
+                    teamLead,
+                    teamMembers,
+                });
+                setFilteredMembers(teamMembers);
+                setIsLoading(false);
+            } catch (error) {
+                Error('Error fetching team users:', error);
```

Comment:
Adds error handling for logging any errors that occur during the data fetching process.